aclk/mqtt.c

```diff
@@ -226,7 +260,69 @@ int _link_lib_init(char *aclk_hostname, int aclk_port, void (*on_connect)(void *
     return rc;
 }
 
-int _link_event_loop(int timeout)
+static inline void _link_mosquitto_write()
+{
+    if(!mqtt_over_websockets)
+        return;
+
+    mosquitto_loop_misc(mosq);
+    if(likely( mosquitto_want_write(mosq) ))
+        mosquitto_loop_write(mosq, 1);
+}
+
+void aclk_lws_connect_notif_callback(){
+    //the connection is done by LWS so this parameters dont matter
+    //ig MQTT over LWS is used
+    _link_mqtt_connect("doesntmatter", 12345);
+    _link_mosquitto_write();
+}
+
+void aclk_lws_data_received_callback(){
+    mosquitto_loop_read(mosq, 1);
```

Comment:
It is the third function that returns integer and you are not getting it to test, is there a special motive for this?

Reply:
well if it fails it will call disconnect callback anyway which kind of fits more what I try/need to do. But I will add logs there to make it easier to debug.

Reply:
In this case I call this only in case when there actually is something to read.